### PR TITLE
git: add basic support for delta

### PR DESCRIPTION
**Pull request by marsam**
### Description

Add [`delta`](https://github.com/dandavison/delta) to git

### Checklist

- [X] Code formatted with `./format`.

- [X] Code tested through `nix-shell --pure tests -A run.all`.

- [X] Test cases updated/added. See [example](https://github.com/rycee/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [X] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/rycee/home-manager/blob/master/CONTRIBUTING.md#commits) for more information and [recent commit messages](https://github.com/rycee/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/rycee/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: modules/programs/git.nix [hunk 1]
```diff
@@ -205,6 +205,22 @@ in {
           '';
         };
       };
+
+      delta = {
+        enable = mkEnableOption ''
```
Comment (kalbasit) [inline modules/programs/git.nix:210]: Can you please add this to the news?

File: modules/programs/git.nix [hunk 2]
```diff
@@ -299,5 +315,13 @@ in {
             ([ "git-lfs" "smudge" ] ++ skipArg ++ [ "--" "%f" ]);
         };
     })
+
+    (mkIf cfg.delta.enable {
+      home.packages = [ pkgs.gitAndTools.delta ];
+
+      programs.git.iniContent.core = {
+        pager = concatStringsSep " " ([ "delta" ] ++ cfg.delta.options);
```
Comment (Kha) [inline modules/programs/git.nix:323]: Wouldn't it be a bit cleaner to point directly into the nix store here (i.e. `[ pkgs.gitAndTools.delta ]`) instead extending `home.packages`? I assume that most users do not need `delta` in their `PATH`, perhaps in contrast to `git lfs` configured in the same way.

File: modules/programs/git.nix (hunk 1 above)
Comment (marsam) [reply modules/programs/git.nix:220]: sure, I added to and entry

File: modules/programs/git.nix (hunk 2 above)
Comment (marsam) [reply modules/programs/git.nix]: fair enough, I've removed delta from `home.packages`

**Comment by rycee** [comment]
Thanks! Rebased to master in cba7b6ee6e056421f862b008b45f1ff9cc2e7252.

**Pull request closed without merging**